### DISCUSSION TITLE
Load audio waveform, but load the audio file ONLY when the user clicks 'Play' button?

### DISCUSSION THREAD

**Discussion by danielmat3** [katspaugh/wavesurfer.js#3578] [Q&A]
Hello!

How do I load just the audio waveform and make it so the audio file downloads only when the user clicks 'Play'? I'm not sure how to set the 'Preload' option to "none".

I've researched online on many forums from other people, but there's no explanation for that... ?

**Specifically I'm looking how to set the audio settings to preload="none".**

Similar discussions:
1. https://github.com/katspaugh/wavesurfer.js/discussions/2752
2. https://github.com/katspaugh/wavesurfer.js/issues/1237
3. (Take a look at this one):
https://stackoverflow.com/questions/38918909/how-can-i-load-only-waveform-and-wait-to-user-click-play-to-download-the-audio

My website where I want to solve the problem:
https://unwav.com/product/groove-premium-tech-house-sample-pack/

I have integrated WaveSurfer-WP plugin from X-Raym:
https://wordpress.org/plugins/wavesurfer-wp/

The following plugin uses your WaveSurfer.js script but from how much I can see it is "outdated". I reached out to X-Raym (plugin creator) and he told me that I need to edit the wavesurfer.js parameters to achieve the result.

=========

There's a similar website that uses the same WaveSurfer-WP plugin like me and integrated CDN for the audio files, but if you compare his wavesurfer.js & wavesurfer-wp.js files the code is nothing similar to mine.

Please see: https://stickz.co/sentio/

On this website above ^ he managed to make this work and the audio is set to - preload="none"

Since you are the "god" of wavesurfer and you know what you are doing, I am reaching out to you for a solution for this...

![Q1](https://github.com/katspaugh/wavesurfer.js/assets/162363136/963b81ae-8bfd-4533-9649-8991aa39242d)

**Comment by katspaugh** [comment]
Wavesurfer needs to download the entire audio to render the waveform. It then also uses the downloaded blob for playback. The only way to avoid loading the audio is to pre-decode peaks:

```js
const media = new Audio()
media.preload = 'none'

const wavesurfer = WaveSurfer.create({
  container: '..,',
  media,
  peaks: [ 0.1, 0.2, 0.3 ],
  duration: 123,
})
```

**Reply by danielmat3** [reply]
Hey Katspaugh!

Thank you so much for taking the time to check this out! 😊

Could you please help me out by letting me know where I should paste the code you sent?

Should I paste it in the **wavesurfer.js** file or **wavesurfer-wp.js** file? If so... can I paste it anywhere or is there a specific line so I don't break the code?

![Q2](https://github.com/katspaugh/wavesurfer.js/assets/162363136/dd69aa00-f205-41bd-a4e6-098b90bb7130)

Also, if we manage to fix this, I'd be thrilled to send you a donation as a token of my appreciation for your help! Thank you again!

**Reply by katspaugh** [reply]
To clarify, wavesurfer **HAS TO download the entire audio** to render the waveform. There's no point adding preload=none if you aren't pre-decoding the audio data (peaks) which I don't think you do.

**Reply by danielmat3** [reply]
Thank you for clarifying, but how can I pre-decode the audio data (peaks)?

**Reply by katspaugh** [reply]
With tools like https://github.com/bbc/audiowaveform or ffmepg.

**Reply by danielmat3** [reply]
What about this code you sent?

const media = new Audio()
media.preload = 'none'

const wavesurfer = WaveSurfer.create({
  container: '..,',
  media,
  peaks: [ 0.1, 0.2, 0.3 ],
  duration: 123,
})

What's it's purpose?

**Reply by katspaugh** [reply]
`peaks: [ 0.1, 0.2, 0.3 ],` is where you insert the pre-decoded peaks.

**Reply by danielmat3** [reply]
Do you have any recommendations how can I implement wavesurfer on wordpress?

**Reply by katspaugh** [reply]
You can use this plugin: https://gist.github.com/katspaugh/5644c7bcf8974ab2e55fb4d7c4f41437
Modify the wavesurfer options directly in the code. Use like this:

```
[wavesurfer url="..."]
```

**Reply by danielmat3** [reply]
That does not seem to work for me, look at the screenshot

![kats plugin](https://github.com/katspaugh/wavesurfer.js/assets/162363136/84d66131-19ef-4e73-93e1-4627a8c561f5)

[wavesurfer url="https://unwav.b-cdn.net/GROOVE_Samples/Unwav_Groove_Top_Loop_10_124BPM.mp3"]

Website page:
https://stg-unwavcom-staging.kinsta.cloud/product/groove-premium-tech-house-sample-pack/

![wavce](https://github.com/katspaugh/wavesurfer.js/assets/162363136/e063c627-1262-4d95-a1f3-5757b7127ffd)

**Reply by katspaugh** [reply]
The shortcode should be inside a blog post, as a part of the content.

**Reply by danielmat3** [reply]
Yes, but it would be better to have Wavesurfer audio on my product pages. Since my products are related to music production, having audio demos directly on the product pages would offer customers a clearer preview of what they can expect.

Here's an example from a different website:
https://cymatics.fm/collections/store/products/mythic-hyperpop-melody-collection

**Comment by theDanielJLewis** [comment]
Is it possible to _replace_ the prerendered peaks when the audio file _does_ download?

**Reply by katspaugh** [reply]
You’d have to call load twice in that case. First with peaks passed and a second time without.
